src/types/type_param.rs

```diff
@@ -33,14 +33,21 @@ pub enum TypeParam {
     Extensions,
 }
 
+impl TypeParam {
+    /// [`TypeParam::BoundedUSize`] with the maximum bound (`u64::MAX`)
+    pub const fn max_usize() -> Self {
+        Self::BoundedUSize(u64::MAX)
+    }
+}
+
 /// A statically-known argument value to an operation.
 #[derive(Clone, Debug, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
 #[non_exhaustive]
 pub enum TypeArg {
     /// Where the (Type/Op)Def declares that an argument is a [TypeParam::Type]
     Type(Type),
-    /// Instance of [TypeParam::USize]. 64-bit unsigned integer.
-    USize(u64),
+    /// Instance of [TypeParam::BoundedUSize]. 64-bit unsigned integer.
+    BoundedUSize(u64),
```

Comment:
I'd be OK with that.

The other thing that slightly grates is that the parameter is the maximum allowed value rather than the minimum excluded value, which would be more consistent with range conventions and also allow us to define an instance with 0 values, but if we made that change we'd either have to make the parameter a `u128` or not allow 2^64-1 as a value; not sure I like either of those options though both probably harmless.

Reply:
yes I only made that choice for that reason. I don't like not allowing 2^64 -1, that could lead to surprising failures. making the parameter a u128 is probably fine though.

Reply:
going the whole hog in commit [d13a4c1](https://github.com/CQCL-DEV/hugr/pull/445/commits/d13a4c112165f2a65e2ed93099e3dbe9d82f25dc) to use an Option<NonZeroU64> to encode the bound in the same size as a u64.